homeassistant/components/sms/sensor.py

```diff
@@ -3,59 +3,46 @@
 
 import gammu  # pylint: disable=import-error
 
-from homeassistant.components.sensor import SensorEntity
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
 from homeassistant.const import DEVICE_CLASS_SIGNAL_STRENGTH, SIGNAL_STRENGTH_DECIBELS
 
 from .const import DOMAIN, SMS_GATEWAY
 
 _LOGGER = logging.getLogger(__name__)
 
+SENSOR_DESCRIPTION = {
+    "signal": SensorEntityDescription(
+        key="signal",
+        name="GSM Signal IMEI",
+        device_class=DEVICE_CLASS_SIGNAL_STRENGTH,
+        unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
+        entity_registry_enabled_default=False,
+    )
+}
+
 
 async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the GSM Signal Sensor sensor."""
     gateway = hass.data[DOMAIN][SMS_GATEWAY]
-    entities = []
     imei = await gateway.get_imei_async()
-    name = f"gsm_signal_imei_{imei}"
-    entities.append(
-        GSMSignalSensor(
-            hass,
-            gateway,
-            name,
-        )
-    )
-    async_add_entities(entities, True)
+    async_add_entities([GSMSignalSensor(hass, gateway, imei, SENSOR_DESCRIPTION)], True)
```

Comment:
```suggestion
    async_add_entities([GSMSignalSensor(hass, gateway, imei, SensorEntityDescription(
        key="signal",
        name="GSM Signal IMEI",
        device_class=DEVICE_CLASS_SIGNAL_STRENGTH,
        unit_of_measurement=SIGNAL_STRENGTH_DECIBELS,
        entity_registry_enabled_default=False,
    ))], True)
```
I doubt black will like this format, but you get the idea